sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveParquetSuite.scala

```diff
@@ -168,4 +170,23 @@ class HiveParquetSuite extends QueryTest with ParquetTest with TestHiveSingleton
       }
     }
   }
+
+  test("SPARK-37210: Concurrent write to different static partitions") {
```

Comment:
This PR aims to support concurrent write for `INSERT OVERWRITE` only. Do we support non-overwrite INSERT case?

Reply:
> non-overwrite

Concurrent write in `INSERT INTO` mode seems to have the same problem, but this pr does not fix it.

Reply:
Do we have a JIRA issue for that? If not, could you file one?